src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/KeyFrameEvaluator.cs

```diff
@@ -80,13 +80,15 @@ public void PersistPriorFrameStatus()
          * <summary>Publicly callable.. caches the statuses of the last passed key frame for computing delta counts from</summary>
          */
 
-        public bool Matched(bool firstSegment, int segmentNumber, List<KeyFrameCriteria> criteriaList)
+        public bool Matched(bool firstSegment, BotSegment botSegment)
```

Comment:
This method accepting BotSegment seems fine/good, but MatchedHelper and other methods and evaluator class methods should only get the segmentNumber like before these changes as noted in my other comment.